### PR TITLE
test(perps): add transfer unit tests

### PR DESCRIPTION
**Pull request by omri-ba-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-perpetual/235)

---

> [!NOTE]
> **Low Risk**
> Test-only changes that add coverage for transfer edge cases; no production contract logic is modified.
> 
> **Overview**
> Expands transfer test coverage in `transfer_tests.cairo` by adding scenarios for transferring a *pending* spot asset balance and for rejecting transfers of a non-existent `AssetId` (`NO_SUCH_ASSET`).
> 
> Adds a flow test ensuring protocol/vault positions cannot receive vault-share assets via spot transfer, expecting `VAULT_CANNOT_HOLD_SHARES`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c461b7dd9861e77b2a6ebb1720a28ff0690027ed. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by omri-ba-starkware** [comment]
* **#235** <a href="https://app.graphite.com/github/pr/starkware-libs/starknet-perpetual/235?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/starkware-libs/starknet-perpetual/235?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `dev`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo [hunk 1]
```diff
@@ -174,3 +178,117 @@ fn test_unsuccessful_transfer_of_vault_share_not_enough_balance() {
...
+    state
+        .transfer(
+            :operator_nonce,
+            asset_id: collateral_id,
+            recipient: transfer_args.recipient,
+            position_id: transfer_args.position_id,
+            amount: transfer_args.amount,
+            expiration: transfer_args.expiration,
+            salt: transfer_args.salt,
+        );
```
[the remark this answers is not in the document]
Comment (omri-ba-starkware) [reply workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo:237]: This should pass, TODO the same in withdraw

File: workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo [hunk 2]
```diff
@@ -174,3 +178,117 @@ fn test_unsuccessful_transfer_of_vault_share_not_enough_balance() {
         );
 }
 
+#[test]
+fn test_transfer_inactive_asset() {
+    // Setup:
+    let cfg: PerpetualsInitConfig = Default::default();
+    let token_state = cfg.collateral_cfg.token_cfg.deploy();
+    let mut state = setup_state_with_pending_spot_asset(cfg: @cfg, token_state: @token_state);
```
[the remark this answers is not in the document]
Comment (omri-ba-starkware) [reply workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo:186]: Fixed

**Review by omri-ba-starkware — COMMENTED** [review]
> @omri-ba-starkware made 1 comment and resolved 1 discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-perpetual/235)* status: 0 of 2 files reviewed, all discussions resolved (waiting on @MohammadNassar1).

**Review by MohammadNassar1 — COMMENTED** [review]
> @MohammadNassar1 reviewed 1 file and all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-perpetual/235)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @omri-ba-starkware).

File: workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo [hunk 3]
```diff
@@ -174,3 +178,117 @@ fn test_unsuccessful_transfer_of_vault_share_not_enough_balance() {
...
+        .transfer(
+            :operator_nonce,
+            asset_id: collateral_id,
+            recipient: transfer_args.recipient,
+            position_id: transfer_args.position_id,
+            amount: transfer_args.amount,
+            expiration: transfer_args.expiration,
+            salt: transfer_args.salt,
+        );
+}
```
[the remark this answers is not in the document]
Comment (MohammadNassar1) [reply workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo:250]: @omri-ba-starkware , please fix.

**Review by MohammadNassar1 — CHANGES_REQUESTED** [review]
> @MohammadNassar1 made 1 comment.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-perpetual/235)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @omri-ba-starkware).

File: workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo (hunk 3 above)
[the remark this answers is not in the document]
Comment (omri-ba-starkware) [reply workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo:250]: This is intentional

[the remark this answers is not in the document]
Comment (omri-ba-starkware) [reply workspace/apps/perpetuals/contracts/src/tests/unit_tests/transfer_tests.cairo:250]: Done.

**Review by omri-ba-starkware — COMMENTED** [review]
> @omri-ba-starkware made 1 comment.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-perpetual/235)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @MohammadNassar1).

**Review by MohammadNassar1 — APPROVED** [review]
:lgtm:

> @MohammadNassar1 reviewed 2 files and all commit messages, made 1 comment, and resolved 1 discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-perpetual/235#-:-OlGYMjr2nz3uKvA3DUZ:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @omri-ba-starkware).
